src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -0,0 +1,1110 @@
+                this.vertSelection.axisType = AxisType.clinicalAttribute;
+                break;
+            case EventKey.horz_logScale:
+                this.horzSelection.logScale = !this.horzSelection.logScale;
+                break;
+            case EventKey.vert_logScale:
+                this.vertSelection.logScale = !this.vertSelection.logScale;
+                break;
+            case EventKey.utilities_viewCopyNumber:
+                this.viewType = ViewType.CopyNumber;
+                break;
+            case EventKey.utilities_viewMutationType:
+                this.viewType = ViewType.MutationType;
+                break;
+        }
+    }
+
+    private setSearchCaseInput(e:any) {
+        this.searchCaseInput = e.target.value;
+        clearTimeout(this.searchCaseTimeout);
+        this.searchCaseTimeout = setTimeout(()=>this.executeSearchCase(this.searchCaseInput), searchInputTimeoutMs);
+    }
+
+    private setSearchMutationInput(e:any) {
+        this.searchMutationInput = e.target.value;
+        clearTimeout(this.searchMutationTimeout);
+        this.searchMutationTimeout = setTimeout(()=>this.executeSearchMutation(this.searchMutationInput), searchInputTimeoutMs);
+    }
+
+    private executeSearchCase(caseId:string) {
+        this.searchCase = caseId;
+    }
+
+    private executeSearchMutation(proteinChange:string) {
+        this.searchMutation = proteinChange;
+    }
+
+    private getHorizontalAxisMenu() {
+        return this.getAxisMenu(false);
+    }
+
+    private getVerticalAxisMenu() {
+        return this.getAxisMenu(true);
+    }
+
+    @action
+    private onGeneLockClick(vertical:boolean) {
+        this.geneLock = !this.geneLock;
+        if (vertical) {
+            this.horzSelection.entrezGeneId = this.vertSelection.entrezGeneId;
+        } else {
+            this.vertSelection.entrezGeneId = this.horzSelection.entrezGeneId;
+        }
+    }
+
+    private onVerticalAxisGeneLockClick() {
+        this.onGeneLockClick(true);
+    }
+
+    private onHorizontalAxisGeneLockClick() {
+        this.onGeneLockClick(false);
+    }
+
+    @action
+    private onGeneSelect(vertical:boolean, option:any) {
+        let targetSelection;
+        let otherSelection;
+        if (vertical) {
+            targetSelection = this.vertSelection;
+            otherSelection = this.horzSelection;
+        } else {
+            targetSelection = this.horzSelection;
+            otherSelection = this.vertSelection;
+        }
+        targetSelection.entrezGeneId = option.value;
+        if (this.geneLock) {
+            otherSelection.entrezGeneId = option.value;
+        }
+    }
+
+    private onVerticalAxisGeneSelect(option:any) {
+        this.onGeneSelect(true, option);
+    }
+
+    private onHorizontalAxisGeneSelect(option:any) {
+        this.onGeneSelect(false, option);
+    }
+
+    @computed get geneOptions() {
+        if (this.props.store.genes.isComplete) {
```

Comment:
i think it would be better to check load state in view rather than in computed.   that way status isn't hidden.  components which consume this data should handle loadind status